bin/workflows-fetch.rb

```diff
@@ -4,34 +4,88 @@
 require 'uri'
 require 'yaml'
 
-# Get the list of workflows
-def fetch_workflows(server)
-  uri = URI.parse("#{server}/api/workflows/")
+def request(url)
+  uri = URI.parse(url)
   request = Net::HTTP::Get.new(uri)
+  request['Accept'] = 'application/json'
   req_options = {
     use_ssl: uri.scheme == 'https',
   }
   response = Net::HTTP.start(uri.hostname, uri.port, req_options) do |http|
     http.request(request)
   end
+  response
+end
+
+# Get the list of workflows
+def fetch_workflows(server)
+  response = request("#{server}/api/workflows/")
 
   JSON.parse(response.body).map do |w|
     w['server'] = server
     w
   end
 end
 
+def fetch_workflowhub()
+  response = request("https://workflowhub.eu/workflows?filter[workflow_type]=galaxy")
+  data = JSON.parse(response.body)
+  if !data['links']['next'].nil?
+    puts "ERROR: Cannot yet handle multiple pages"
+    exit 42
+  end
+  puts "INFO: Fetching #{data['data'].length} workflows from WorkflowHub"
+  data['data'].map.with_index { |w, i|
+    # {"id"=>"14", "type"=>"workflows", "attributes"=>{"title"=>"Cheminformatics - Docking"}, "links"=>{"self"=>"/workflows/14"}}
+    wf_info = JSON.parse(request("https://workflowhub.eu#{w['links']['self']}").body)
+    creator0 = wf_info['data']['attributes']['creators'][0]
```

Comment:
I think `<div class="rli-project-list hidden-xs">` would be good to parse, too, since it has the Teams information, e.g. iwc.

Reply:
I don't have easy access to that in the endpoint I'm working with currently. We do see this, but that requires additional HTTP requests to fetch. So currently we look only at creators + other_creators which are mostly filled out.

```
      "creators": [],
      "other_creators": "Wolfgang Maier",
...
    "relationships": {
      "creators": {
        "data": []
      },
      "submitter": {
        "data": [
          {
            "id": "112",
            "type": "people"
          }
        ]
      },
      "people": {
        "data": [
          {
            "id": "112",
            "type": "people"
          }
        ]
      },
      "projects": {
        "data": [
          {
            "id": "33",
            "type": "projects"
          }
        ]
      },
```

Reply:
It's wildly denormalised, everything takes 10 requests for some reason. But yes! that's a good suggestion, that's only 1 then.

Shall I just comma-separate that with whatever is in `creators` and `other_creators` and the `relationships.projects.data[]` then? There are so many fields for attribution here that it seems silly somehow to stuff so many names in the page (especially compared to Galaxy workflows with only a single owner)

Reply:
Thanks, good suggestions, will implement.

Reply:
Implemented.